### PR TITLE
Add TextDelegateDecoder to standard Feign clients

### PR DESCRIPTION
**Pull request by uschi2000**
@markelliot

### CODE REVIEW

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java [hunk 1]
```diff
@@ -67,9 +68,9 @@ public static FeignClientFactory withMapper(ObjectMapper mapper) {
         return FeignClientFactory.of(
                 new JAXRSContract(),
                 new JacksonEncoder(mapper),
-                new OptionalAwareDecoder(new JacksonDecoder(mapper), SerializableErrorErrorDecoder.INSTANCE),
+                new OptionalAwareDecoder(new TextDelegateDecoder(new JacksonDecoder(mapper)),
```
Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java]: is the order here correct?

File: http-clients/src/test/java/feign/TextDelegateDecoderTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        Object decodedObject = textDelegateDecoder.decode(response, String.class);
+
+        assertEquals(decodedObject, DELEGATE_RESPONSE);
+    }
+
+    @Test
+    public void testStandardClientsUseTextDelegateEncoder() {
+        assertThat(service.getString("string"), is("string"));
+
+    }
```
Comment (markelliot) [inline http-clients/src/test/java/feign/TextDelegateDecoderTest.java:142]: can we also cover that a null response comes back with content type text/plain and you get the right `Optional` behavior?

Comment (uschi2000) [inline http-clients/src/test/java/feign/TextDelegateDecoderTest.java:142]: do you mean in interface like this?

```
        @Produces(MediaType.TEXT_PLAIN)
        Optional<String> getString(@QueryParam("value") String value);
```

Not really sure that actually makes sense.

Comment (uschi2000) [inline http-clients/src/test/java/feign/TextDelegateDecoderTest.java:142]: added a few tests to clarify/specify the behaviour.

**Pull request merged**